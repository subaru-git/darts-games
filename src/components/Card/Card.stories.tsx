import type { Meta, StoryObj } from '@storybook/react';
import Card from './Card';

const meta: Meta<typeof Card> = { component: Card, title: 'Components/Card' };

export default meta;
type Story = StoryObj<typeof Card>;
export const Default: Story = { args: { children: <p>The Card</p> } };
