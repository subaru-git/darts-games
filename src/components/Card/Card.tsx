import React, { FC, ReactNode } from 'react';

type CardProps = {
  children?: ReactNode;
};

const Card: FC<CardProps> = ({ children }) => {
  return (
    <div className='h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md'>
      {children}
    </div>
  );
};

export default Card;
