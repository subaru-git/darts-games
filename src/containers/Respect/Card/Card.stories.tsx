/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Card from './Card';

export default {
  title: 'Containers/Respect/Card',
  component: Card,
} as ComponentMeta<typeof Card>;

const Template: ComponentStory<typeof Card> = (args) => <Card {...args} />;

export const Default = Template.bind({});
Default.args = {
  data: {
    title: 'ダーツライブ 日本 | DARTSLIVE',
    image: 'https://www.dartslive.com/wp-content/themes/dartslive/images/apple-touch-icon.png',
    description:
      'ネットワークとダーツライブカードを使ったダーツ マシンDARTSLIVE(ダーツライブ)のオフィシャルサイト。DARTSLIVE設置店舗(ダーツバー)検索、ダーツライブ2やイベント情報、ダーツライブランキング等。',
    url: 'https://www.dartslive.com/jp/',
    type: 'website',
  },
};
