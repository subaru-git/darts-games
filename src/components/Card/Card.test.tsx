import { render } from '@testing-library/react';
import Card from './Card';

test('should rendering', () => {
  const { container } = render(<Card />);
  expect(container).toMatchSnapshot();
});
