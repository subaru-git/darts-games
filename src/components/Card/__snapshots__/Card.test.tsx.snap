// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
  >
    <p>
      The Card
    </p>
  </div>
</div>
`;
